### PR TITLE
Produce an error at runtime rather than at compile time for unsupport…

### PR DESCRIPTION
**Pull request by bjorn3**
…ed architectures in cranelift-native

### CODE REVIEW

**Review by cfallin — APPROVED** [review]
Thanks! Happy to take this, maybe with the whole block removed instead (see below).

For others' reference, this PR comes from [this](https://github.com/bytecodealliance/wasmtime/pull/5226#issuecomment-1402404126) discussion.

File: cranelift/native/src/lib.rs [hunk 1]
```diff
@@ -48,7 +48,7 @@ pub fn builder_with_options(infer_native_flags: bool) -> Result<isa::Builder, &'
 
     #[cfg(target_arch = "x86")]
     {
-        compile_error!("Unsupported architecture");
+        return Err("unsupported architecture");
```
[the remark this answers is not in the document]
Comment (bjorn3) [reply cranelift/native/src/lib.rs:51]: Right

**Review by cfallin — APPROVED** [review]

**Comment by bjorn3** [comment]
Would a patch release for cranelift-native be possible or should I vendor it for the time being?

**Comment by cfallin** [comment]
I think this probably doesn't meet the bar for a patch release (unsupported platform, no functional correctness issue, just ergonomics around how to fail cleanly); probably best to vendor, sorry! It'll go out with the 6.0 release on Feb 20.

**Pull request merged**